common/static/js/src/ReactRenderer.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+export class ReactRenderer {
+  constructor(component, selector, props) {
+    const path = component;
+    // ComponentToRender = import(path);
+    this.elementList = document.querySelector(selector);
+    this.props = props;
+    import(/* webpackChunkName: "dynamicBundle" */ path)
+      .then(ComponentToImport => {
+        console.log(ComponentToImport);
+        this.renderComponent();
+      });
```

Comment:
This `import()` is giving me grief. It works if `path` is a globally accessible `node_module`, such as `react` or `@edx/paragon`. But I haven't been able to get it to work with any local modules. I found [this tutorial](https://dev.to/kayis/dynamic-imports-with-webpack-2) which was kind of helpful for wrapping my brain around dynamic imports, but I'm not sure its approach will work here at all.

So far I've tried:
* setting `publicPath` to `/static`. This doesn't seem to be a good idea since it affects **ALL** webpack includes.
* using a relative path to `ReactRenderer` for a module in the same directory as `ReactRenderer` (`'./HelloWorld.jsx'`)
* using a path relative to the edx-platform root
* using the path to the on-the-fly generated webpack bundle relative to the edx-platform root (that's what the [`webpack_static`](https://github.com/edx/edx-platform/pull/16639/files#diff-1eb576102f40a534c2862d86fc3ff321R169) call above is doing, but it doesn't work)

I have not tried:
* messing further with `publicPath`
* path relative to the html template where the mako def is being invoked

Reply:
There's also [this example repo](https://github.com/mjaniszew/dynamic-imports-example) but the syntax here did not work for me.

Reply:
Update: I was doing it wrong. [`import(fullyDynamicPath)` does not work in webpack](https://webpack.js.org/api/module-methods/) -- thanks to Matt T for pointing that out. So I'm now trying with `import('./staticPath.jsx')` and it's throwing a not-especially-helpful `Error: Loading chunk 0 failed` which I'm still debugging.